src/main/java/com/google/firebase/auth/AuthProviderConfig.java

```diff
@@ -29,16 +29,22 @@
  */
 public abstract class AuthProviderConfig {
 
-  @Key("name")
+  // Lazily initialized from 'resourceName'.
```

Comment:
May be just remove this property altogether, and always return the calculated value from the getter. It's quite cheap to compute.

Reply:
Okay! For consistency, I also made this change to the Tenant class.